File: ExpressionToString.Tests/CompilerGenerated/Member.cs
using System;
using Xunit;
using static ExpressionToString.Tests.Globals;
using static ExpressionToString.Tests.Runners;

namespace ExpressionToString.Tests {
    [Trait("Source", CSharpCompiler)]
    public class Member {
        [Fact]
        public void InstanceMember() {
            var s = "";
            RunTest(
                () => s.Length,
                "() => s.Length",
                "Function() s.Length"
            );
        }

        [Fact]
        public void ClosedVariable() {
            var s = "";
            RunTest(
                () => s,
                "() => s",
                "Function() s"
            );
        }

        [Fact]
        public void StaticMember() => RunTest(
            () => string.Empty,
            "() => string.Empty",
            "Function() String.Empty"
        );

        [Fact(Skip ="Test for nested closure scopes")]
        public void NestedClosedVariable() {
            Assert.False(true);
        }
    }
}

File: Tests.Shared.VB/CompilerGenerated/Member.vb
Public Class Member
    <Fact>
    Sub InstanceMember()
        Dim s = ""
        RunTest(
            Function() s.Length,
            "() => s.Length",
            "Function() s.Length"
        )
    End Sub

    <Fact>
    Sub ClosedVariable()
        Dim s = ""
        RunTest(
            Function() s,
            "() => s",
            "Function() s"
        )
    End Sub

    <Fact>
    Sub StaticMember()
        RunTest(
            Function() String.Empty,
            "() => string.Empty",
            "Function() String.Empty"
        )
    End Sub

    <Fact(Skip:="Test for nested scope")>
    Sub NestedClosureScope()
        Assert.False(True)
    End Sub
End Class
